Allow cross referencing `icon` and `icons` entries (#2538)
openstreetmap/id-tagging-schema: SCHEMA.md

##### `icons`

On [combo / dropdown](#combodropdown-fields) and [radio](#radio-buttons) fields, the `icons` object maps each option key to an icon id shown beside that value in the editor, e.g. `"zebra": "iD-crossing_markings-zebra"` in the snippet below. More information about available icon sets and usage of icons can be found on the [icons subpage](ICONS.md).

> You may set each value to `"{presets/<preset-id>}"` instead of a literal id, with the same rules as [preset `icon`](#icon).

Reviewer: ```suggestion
You may also instead use `"{presets/<preset-id>}"` references instead of a literal id, with rules as for [preset `icon`](#icon).
```

if we want `presets` prefix then we cannot claim that rules are the same

"may set each value" also sounded like you cannot reference only for some

##### `iconsCrossReference`

> An optional property to copy the entire `icons` object from another field by giving that field's id in brackets (no `presets/` prefix)—for example `{kerb}` copies the field whose id is `kerb`. Useful when several field variants share the same option icons.

Reviewer: ```suggestion
An optional property to reference to the `icons` object of another field, indicated  by using that field's name contained in brackets, like `{field}`. `presets/` prefix is not included - for example `{kerb}` copies the field whose id is `kerb`. This is useful for example when there are multiple variants of fields for the same tag, which should all use the same icons.
```

Trying shorter sentences like it was before. Other changes that were introduced should be preserved, except `—`

Reviewer: are you expecting to have time to look into it?

I am considering making own PR based on this one, or making own one manually, as ideally we would have this functionality to allow making https://github.com/openstreetmap/id-tagging-schema/pull/2633 and similar in a smarter way

> If a field uses `iconsCrossReference`, the builder copies the other field’s `icons` map first, then expands every `{presets/…}` value in **all** field `icons` maps (including the copy).

Reviewer: does order actually matter here?
